### PR TITLE
Adds `notices-as-warnings` option

### PR DESCRIPTION
**Pull request by donatj**
GitHub Actions don't actually have a notice level, so they just get added to the log but not the file tab

image

This PR just adds a `--no-notices` flag that lets you have them as GitHub warnings.

### CODE REVIEW

**Comment by staabm** [comment]
Hmm.. per docs `notice` is a valid annotation level, see https://docs.github.com/en/rest/reference/checks#check-runs

PNG-Bild

**Comment by donatj** [comment]
@staabm  The checks API supports creating notices, but not action commands

https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message

**Comment by donatj** [comment]
There's also debug, but it behaves differently and doesn't have filename/line support.

image

**Comment by donatj** [comment]
Looks like #61 backs me up too.

**Comment by staabm** [comment]
I will have a look later today. Thx for the PR

File: cs2pr [hunk 1]
```diff
@@ -34,6 +35,9 @@ foreach ($argv as $arg) {
         case 'colorize':
             $colorize = true;
             break;
+        case 'no-notices':
```
Comment (staabm) [inline cs2pr:38]: Thx for the PR. I like the change in general.

I feel this option should be named `notices-as-warnings` or similar.
Additionally the new option should be added to the help output.

lastly this PR changes the mode-bits of this file - did this change intentionally?

**Comment by donatj** [comment]
> I feel this option should be named notices-as-warnings or similar.

`notices-as-warnings` sounds good me - naming things is one of the hardest parts of programming 😉 

> a unit test is also mandatory.

Got that added for you.

> lastly this PR changes the mode-bits of this file - did this change intentionally?

I did `chmod +x cs2pr`  so I could run `./cs2pr` while testing my changes. I can revert that if you'd like. However, I think it makes sense for it to be executable given the shebang on line 1

https://github.com/staabm/annotate-pull-request-from-checkstyle/blob/8bc869542d45abb2f0e6f3496ad67dbf9369f9a5/cs2pr#L1-L2

**Comment by donatj** [comment]
Pushed a fix to try to match the coding standard.

**Review by staabm — CHANGES_REQUESTED** [review]
Thanks!

The new option should be added to the help output. 
Otherwise lgtm

**Comment by donatj** [comment]
Fixed the remaining code style error, added help output

File: cs2pr [hunk 2]
```diff
@@ -55,6 +59,7 @@ if (count($params) === 1) {
     echo "Supported options:\n";
     echo "  --graceful-warnings   Don't exit with error codes if there are only warnings.\n";
     echo "  --colorize            Colorize the output (still compatible with Github Annotations)\n";
+    echo "  --notices-as-warnings Convert notices to warnings.\n";
```
Comment (staabm) [inline cs2pr:62]:
```suggestion
    echo "  --notices-as-warnings Convert notices to warnings (Github does not annotate notices otherwise).\n";
```

**Review by staabm — APPROVED** [review]
Love it, thanks

**Pull request merged**